sensor/common/registry/registry_store.go

```diff
@@ -201,3 +213,68 @@ func (rs *Store) GetGlobalRegistryForImage(image *storage.ImageName) (registryTy
 
 	return nil, errors.Errorf("unknown image registry: %q", reg)
 }
+
+// SetDelegatedRegistryConfig sets a new delegated registry config for use in determining
+// if a particular image is from a registry that should be accessed local to this cluster
+func (rs *Store) SetDelegatedRegistryConfig(config *central.DelegatedRegistryConfig) {
+	rs.delegatedRegistryConfigMutex.Lock()
+	defer rs.delegatedRegistryConfigMutex.Unlock()
+	rs.delegatedRegistryConfig = config
+}
+
+// IsLocal determines if an image is from a registry that should be accessed
+// local to this secured cluster.  Always returns true for image registry's that have
+// been added via AddClusterLocalRegistryHost
+func (rs *Store) IsLocal(image *storage.ImageName) bool {
+	if image == nil {
+		return false
+	}
+
+	if rs.hasClusterLocalRegistryHost(image.GetRegistry()) {
+		// This host is always cluster local irregardless of the DelegatedRegistryConfig (ie: OCP internal registry).
+		return true
+	}
+
+	rs.delegatedRegistryConfigMutex.RLock()
+	defer rs.delegatedRegistryConfigMutex.RUnlock()
+
+	config := rs.delegatedRegistryConfig
+	if config == nil || config.EnabledFor == central.DelegatedRegistryConfig_NONE {
+		return false
+	}
+
+	if config.EnabledFor == central.DelegatedRegistryConfig_ALL {
+		return true
+	}
+
+	// if image matches a delegated registry prefix, it is local
+	imageFullName := urlfmt.TrimHTTPPrefixes(image.GetFullName())
+	for _, r := range config.Registries {
+		regPath := urlfmt.TrimHTTPPrefixes(r.RegistryPath)
+		if strings.HasPrefix(imageFullName, regPath) {
+			return true
+		}
+	}
```

Comment:
Shouldn't we also potentially verify whether the `cluster_id` is matching here, or am I missing some context on the delegated registry config?

AFAIU, with the current delegated registry config you can essentially say "images from this registry shall be scanned by this secured cluster". The reasons may be different (such as credentials on the cluster, network separation etc.), shouldn't we also respect this here and skip such configurations in case the `cluster_id` is set?

Reply:
Ah gotcha, makes sense.

Yeah it might be worthwhile to remove the `cluster_id` from the internal API then, or at the very least a comment on the proto, so it's somewhat obvious that it currently does not apply here.